sklearn/model_selection/_split.py

```diff
@@ -298,7 +298,9 @@ def __init__(self, n_splits, shuffle, random_state):
 
         self.n_splits = n_splits
         self.shuffle = shuffle
-        self.random_state = random_state
+        self.random_state = random_state  # only here for repr to work
+        self._seed = _get_seed_from_random_state_param(random_state)
```

Comment:
The fact that this is done in the init breaks our convention. Arguably, these conventions are only for estimators, so it's not "outlawed" :).

The reason behind those conventions is to enable people changing public attributes without having surprising behavior.

If we were to adapt these conventions, we would do this call in the beginning of the "split" method. I haven't given a lot of thoughts to this choice. Do you have an educated opinion of what would be better?